modules/apps/map/map-google-maps/src/main/resources/META-INF/resources/js/MapGoogleMaps.es.js

```diff
@@ -34,6 +34,12 @@ class MapGoogleMaps extends MapBase {
 	constructor(...args) {
 		super(...args);
 
+		MapBase.DialogImpl = GoogleMapsDialog;
```

Comment:
Why are we doing this in the constructor? Seems a bit strange.

Reply:
To update the `MapBase.*Impl` to the current map provider (`MapOpenStreetMap` or `MapGoogleMaps`) when [is created](https://github.com/liferay/liferay-portal/blob/ea44893298b4a31ba1da3206f5a0650e0d124cad/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Geolocation/useGeolocation.es.js#L137-L162).

If we leave it outside the `MapBase.*Impl` are updated when we import it and the last provider imported overwrite the global MapBase no matter if it is used or not. 

Only the last imported module works if we leave it as is. And maybe this work in the past when we only use require one of the implementations like in [@liferay/map-openstreetmap](https://github.com/liferay/liferay-portal/blob/6a38677ec9534829a782d9b1a0961dba4d138dab/modules/apps/map/map-openstreetmap/src/main/resources/META-INF/resources/view.jsp#L38) ([require](https://github.com/liferay/liferay-portal/blob/master/modules/apps/map/map-openstreetmap/src/main/java/com/liferay/map/openstreetmap/internal/OpenStreetMapMapProvider.java#L78)) the same with `@liferay/map-google-maps` in `.jsp`.

Maybe we can move to an `initglobalImpl` method if you mean that 🤔

Reply:
@boton thanks for the information. Now that I've had a closer look at all these files, I understand what you mean. It still feels "strange" and maybe we could have solved that by using composition and not inheritance but, at this point if it works and the echo team is happy, feel free to forward. Thanks.

Reply:
@julien you are right, it feels a little bit strange and took me time to understand how it is working :D

Reply:
gotta love the magic of JavaScript :)